### ISSUE TITLE
Fix false positives for interpolation in property name in custom-property-pattern

### ISSUE THREAD

**Issue by simPod** [stylelint/stylelint#5945]
### What steps are needed to reproduce the bug?

Have this scss

```scss
.a {
    --#{$variable-prefix}table-bg: rgba(#a1f2b0, 0.3);
}
```

### What Stylelint configuration is needed to reproduce the bug?

`stylelint-config-standard`

### How did you run Stylelint?

CLI with `stylelint "**/*.css" --config myconfig.json`

### Which version of Stylelint are you using?

`14.0.0`

### What did you expect to happen?

No errors

### What actually happened?

Triggered `custom-property-pattern rule` error.

### Does the bug relate to non-standard syntax?

scss

### Proposal to fix the bug

Allow variables syntax in `custom-property-pattern` rule

(https://github.com/stylelint/stylelint-config-standard/issues/240)

**Comment by jeddy3** [comment]
Thanks for recreating the report here.

I can reproduce using [the demo](https://stylelint.io/demo#N4Igxg9gJgpiBcIB0BDABMAOgOzWgtPgMTAAkAbigE4CWKARgDYz4AOVMAZjQB4C+AFwbN89AObw0VMfRQAKIigCMnAEz0ADABo0GpAGYAlAG4cfEFpABnAJ7YhPBNbBWrF8BGzcxTrLjSYIFQArsxWgZJ+eHiBYMFWAhAAtmxUEKwwVAI2bCgCApnYEWgA2jjR0YEAenIlKPgAXgC6dY0a+ACcTQBUhnL4rQ3tXQDUht2kgVrlFRgzswEgSTCuKGIwxYEAojwZYAVQaHEJyWjs6ZnZaNgoy2iJaPQwaADWMLL0+GAoVhsg89E+ACmjMgdhzHwgA).

We can to use [`isStandardSyntaxProperty`](https://github.com/stylelint/stylelint/blob/main/lib/utils/isStandardSyntaxProperty.js) (on line 56) before testing if it's a custom property:

https://github.com/stylelint/stylelint/blob/9a0ab5b86845c3a950deaff22ae8c02c71c42e1f/lib/rules/custom-property-pattern/index.js#L56-L59

We can add a test to:

https://github.com/stylelint/stylelint/blob/9a0ab5b86845c3a950deaff22ae8c02c71c42e1f/lib/rules/custom-property-pattern/__tests__/index.js#L129-L142

I've labelled the issue as ready to implement. Please consider [contributing](https://stylelint.io/contributing) if you have time.

There are [steps on how to fix a bug in a rule](https://stylelint.io/developer-guide/rules#fix-a-bug-in-a-rule) in the Developer guide.

**Comment by kawaguchi1102** [comment]
Hey @jeddy3, can I work on this issue?

**Comment by simPod** [comment]
@jeddy3 thanks for pointers. I might look at it but certainly not now, I have still few OSS todo's. In few weeks sure.

**Comment by jeddy3** [comment]
> Hey @jeddy3, can I work on this issue?

@kawaguchi1102 Yes, please do. That'll be great.

> @jeddy3 thanks for pointers. I might look at it but certainly not now

@simPod No worries at all. It looks like @kawaguchi1102 is happy to work on it.

**Comment by kawaguchi1102** [comment]
@jeddy3 thanks!

@simPod I'll take care of this issue!
